packages/polaris-viz/src/components/Grid/GroupCell.tsx

```diff
@@ -0,0 +1,240 @@
+import React, {useEffect, useState} from 'react';
+import type {ScaleLinear} from 'd3-scale';
+
+import styles from './Grid.scss';
+
+interface GroupCellProps {
+  group: CellGroup;
+  index: number;
+  xScale: ScaleLinear<number, number>;
+  cellHeight: number;
+  cellWidth: number;
+  isSmallContainer: boolean;
+  animationStarted: boolean;
+  hoveredGroups: Set<string>;
+  handleGroupHover: (group: CellGroup | null, event: React.MouseEvent) => void;
+  getColors: (group: CellGroup | null) => {bgColor: string; textColor: string};
+  containerWidth: number;
+  isAnimated: boolean;
+}
+
+interface CellGroup {
+  start: {row: number; col: number};
+  end: {row: number; col: number};
+  bgColor: string;
+  color: string;
+  name: string;
+  value: string;
+  secondaryValue: string;
+}
+
+export const GroupCell: React.FC<GroupCellProps> = ({
+  group,
+  index,
+  xScale,
+  cellHeight,
+  cellWidth,
+  isSmallContainer,
+  animationStarted,
+  hoveredGroups,
+  handleGroupHover,
+  getColors,
+  containerWidth,
+  isAnimated,
+}) => {
+  const [isFadingIn, setIsFadingIn] = useState(true);
```

Comment:
not sure why you need to know if it is fading in or not 🤔 could you just add the `FadeIn` class with the animation with `forwards` as you already have it and that should be enough or maybe i'm missing some cases?

Reply:
we can remove this!